nativelib/src/main/resources/gc/commix/LargeAllocator.c

```diff
@@ -0,0 +1,375 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "LargeAllocator.h"
+#include "utils/MathUtils.h"
+#include "Object.h"
+#include "State.h"
+#include "Sweeper.h"
+#include "Log.h"
+#include "headers/ObjectHeader.h"
+
+inline static int LargeAllocator_sizeToLinkedListIndex(size_t size) {
+    assert(size >= MIN_BLOCK_SIZE);
+    assert(size % MIN_BLOCK_SIZE == 0);
+    int index = size / MIN_BLOCK_SIZE - 1;
+    assert(index < FREE_LIST_COUNT);
+    return index;
+}
+
+Chunk *LargeAllocator_chunkAddOffset(Chunk *chunk, size_t words) {
+    return (Chunk *)((ubyte_t *)chunk + words);
+}
+
+void LargeAllocator_freeListPush(FreeList *freeList, Chunk *chunk) {
+    Chunk *head = (Chunk *)freeList->head;
+    do {
+        // head will be replaced with actual value if
+        // atomic_compare_exchange_strong fails
+        chunk->next = head;
+    } while (!atomic_compare_exchange_strong(&freeList->head, (word_t *)&head,
+                                             (word_t)chunk));
+}
+
+// This could suffer from the ABA problem. However, during a single phase each
+// BlockMeta is removed no more than once. It would need to be swept before
+// re-use.
+Chunk *LargeAllocator_freeListPop(FreeList *freeList) {
+    Chunk *head = (Chunk *)freeList->head;
+    word_t newValue;
+    do {
+        // head will be replaced with actual value if
+        // atomic_compare_exchange_strong fails
+        if (head == NULL) {
+            return NULL;
+        }
+        newValue = (word_t)head->next;
+    } while (!atomic_compare_exchange_strong(&freeList->head, (word_t *)&head,
+                                             newValue));
+    return head;
+}
+
+Chunk *LargeAllocator_freeListPop_OnlyThread(FreeList *freeList) {
```

Comment:
Second underscore in the name is not matching our naming convention. Please remove it.

Reply:
Removed